### ISSUE TITLE
Advanced-ControlNet  inpaint  not  working  ?

### ISSUE THREAD

**Issue by bihailantian655** [Kosinkadink/ComfyUI-Advanced-ControlNet#91]
![workflow (1)](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/35267721/e9f956b6-9319-4594-a9e0-75f321ce5896)

The difference between these two workflows is that the first uses native Control Net ,    the  second  use ADV controlnet , and it seems that ADV  inpaint  has very little impact.     

Maybe it's because I'm not using adv cnet correctly, 

![workflow](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/35267721/33edd1af-6f4b-4f1f-8451-e23353d32771)

**Comment by bihailantian655** [comment]
![workflow (2)](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/35267721/f5b1668d-d157-4d53-881d-28d20c059e70)

**Comment by Kosinkadink** [comment]
To reiterate, the mask_optional input on Apply Advanced ControlNet has nothing to do with inpaint or inpaint controlnet. You still have to use the preprocessor same as before.

**Comment by bihailantian655** [comment]
![workflow (3)](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/assets/35267721/98381b98-cfd6-45e4-924c-ea5980791938)

**Comment by Kosinkadink** [comment]
Yes, but you're missing the inpaint preprocessor for the control net, which you had set up correctly for your vanilla controlnet example in your original post.

**Comment by bihailantian655** [comment]
This time it should be right, thank you again for your help

**Comment by Kosinkadink** [comment]
No problem!

**Comment by bihailantian655** [comment]
![Uploading ComfyUI_02520_.png…]()
